contracts/adapters/swap/astroport/src/contract.rs

```diff
@@ -322,6 +339,127 @@ fn query_simulate_swap_exact_asset_out(
     Ok(asset_in_needed)
 }
 
+// Queries the astroport router contract to simulate a swap exact amount in with spot price
+fn query_simulate_swap_exact_asset_in_with_spot_price(
```

Comment:
Made a alteration here for how I think this code should be organzied

https://gist.github.com/zrbecker/dccb56a2a81f4fc0be6ab4fb9d931d09

If these methods are used internally, we should probably the logic into another function `simulate_swap_exact_asset_in`, which `query_simulate_swap_exact_asset_in` and this method can call.

Reply:
completed my take of this new design here: https://github.com/skip-mev/skip-api-contracts/pull/77/commits/1c28a99c18fc6446f973d69df153fdf4a948bdc9

In particular, instead of always giving back all metadata fields, the metadata queries accept "include_X" boolean params that are checked and used to build up the response as needed.